### PR TITLE
[RORDEV-314] Fixed config file permission issue with JDK 1.8 25.262-b10

### PR DESCRIPTION
**Pull request by pielas**
🐞Fix (ES) Config file permission issue with JDK 1.8.0 25.262-b10

### CODE REVIEW

File: es66x/build.gradle [hunk 1]
```diff
@@ -53,7 +53,7 @@ compileScala {
 dependencies {
     compile project(path: ':core')
     compile             group: 'org.elasticsearch',                 name: 'elasticsearch' ,             version: esVersion
-//    compileClasspath    group: 'org.codelibs.elasticsearch.lib',    name: 'plugin-classloader',         version: esVersion // no 6.8.11&12 in mvn atm
```
Comment (coutoPL) [inline es66x/build.gradle:56]: I think you cannot uncomment it. That's why pipeline failed. There are no these two version published. This dependency is only needed to run the project with Idea.

File: core/src/main/scala/tech/beshu/ror/configuration/EsConfig.scala [hunk 2]
```diff
@@ -38,7 +39,7 @@ object EsConfig {
           (implicit envVarsProvider:EnvVarsProvider): Task[Either[LoadEsConfigError, EsConfig]] = {
     val configFile = File(s"${esConfigFolderPath.toAbsolutePath}/elasticsearch.yml")
     (for {
-      _ <- EitherT.fromEither[Task](Either.cond(configFile.exists, (), FileNotFound(configFile)))
+      _ <- EitherT.fromEither[Task](Either.cond(doPrivileged(configFile.exists), (), FileNotFound(configFile)))
```
Comment (coutoPL) [inline core/src/main/scala/tech/beshu/ror/configuration/EsConfig.scala:42]: I'm little bit concerned that we should add `doPrivileged` in so many places. What about creating `File` adapter (I wanted to write decorator, but there is no sense to expose all methods from `better.file.File` class) and encapsulate `doPrivileged` in it?

File: eshome/plugins/readonlyrest/plugin-security.policy [hunk 3]
```diff
@@ -9,4 +9,6 @@ grant {
   permission java.io.FilePermission "/proc/sys/net/core/somaxconn","read";
   permission java.lang.RuntimePermission "accessClassInPackage.sun.misc";
   permission java.lang.RuntimePermission "setContextClassLoader";
+  permission java.lang.RuntimePermission "createSecurityManager";
```
Comment (coutoPL) [inline eshome/plugins/readonlyrest/plugin-security.policy:12]: why this privilege is needed?

**Review by coutoPL — CHANGES_REQUESTED** [review]

File: es66x/build.gradle (hunk 1 above)
Comment (pielas) [reply es66x/build.gradle:56]: aah ok, I it didn't work in idea without that and I thought that they are available.

File: eshome/plugins/readonlyrest/plugin-security.policy (hunk 3 above)
Comment (pielas) [reply eshome/plugins/readonlyrest/plugin-security.policy:12]: It turned out that these two permissions are needed when I removed using `sec-all.policy` which has `permission java.security.AllPermission;`. All of permissions from `plugin-security.policy` are not needed when `sec-all.policy` is used

**Review by coutoPL — APPROVED** [review]
LGTM

**Pull request merged**